tests/e2e/lib/flakes.go

```diff
@@ -48,15 +49,18 @@ func CheckIfFlakeOccured(input string, knownFlake *bool) {
 			StringSearchPattern: "Error copying image: writing blob: uploading layer chunked: received unexpected HTTP status: 500 Internal Server Error",
 		},
 	}
+	logString := strings.Join(*logs, "\n")
 
 	for _, pattern := range flakePatterns {
 		re := regexp.MustCompile(pattern.StringSearchPattern)
-		if re.MatchString(input) {
+		if re.MatchString(logString) {
 			log.Printf("FLAKE DETECTION: Match found for issue %s: %s\n", pattern.Issue, pattern.Description)
 			*knownFlake = true
+			*logs = nil
```

Comment:
Won't this affect the original accumulatedTestLogs ?

Reply:
uhum, but that the idea

after checking for flakes, delete old logs so they do not get mixed with next run logs

Reply:
I think it should not be a hidden behavior of flake function, the caller should be responsible of clearing the logs after the call imo.

Reply:
+1 lets clear the logs outside the flake detection function if needed

Reply:
we already change the value of `knownFlake`, why not change both inside the function?

Reply:
This in my opinion is different. The function is designed to detect flakes in the provided logs, so the consumer of this function intentionally passes a `flip` flag to indicate whether a flake was found.

The user passes as well the logs to be checked for and current behavior may be very unexpected when the logs in which we wanted to find out if the flake occured are modified.

If you really want to do this way I would rename the function to:

`EmptyLogsOnFlakeOccurance`

But that is not really what we want. We could also skip passing the flag and simply return true/false when the flake is found/not found from that function, but having pointer to the variable is easier to manage, because the way tests are invoked.

Reply:
alright, changed function to address comment